dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/provider/DataSourceClientProvider.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.dao.provider;
+
+import static org.apache.dolphinscheduler.common.enums.DbType.HIVE;
+
+import org.apache.dolphinscheduler.common.datasource.BaseConnectionParam;
+import org.apache.dolphinscheduler.common.enums.DbType;
+import org.apache.dolphinscheduler.common.exception.BaseException;
+import org.apache.dolphinscheduler.common.utils.ClassLoaderUtils;
+import org.apache.dolphinscheduler.common.utils.ReflectionUtils;
+import org.apache.dolphinscheduler.common.utils.StringUtils;
+import org.apache.dolphinscheduler.dao.datasource.DataSourceClient;
+import org.apache.dolphinscheduler.dao.datasource.JdbcDriverManager;
+
+import java.io.FilenameFilter;
+import java.net.MalformedURLException;
+import java.sql.Driver;
+import java.util.HashSet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Sets;
+
+/**
+ * DataSource Plug in dynamic loading
+ */
+public class DataSourceClientProvider {
+
+    private static final Logger logger = LoggerFactory.getLogger(DataSourceClientProvider.class);
+
+    private static final JdbcDriverManager jdbcDriverManagerInstance = JdbcDriverManager.getInstance();
+
+    public static DataSourceClient createDataSourceClient(BaseConnectionParam baseConnectionParam, DbType dbType) {
+        logger.info("Creating the createDataSourceClient. JdbcUrl: {} ", baseConnectionParam.getJdbcUrl());
+        //Check jdbc driver location
+        checkDriverLocation(baseConnectionParam, dbType);
```

Comment:
It is better to use the spi module's plug-in loading method uniformly

Reply:
Thank you for your suggestion. I will try to use the SPI module's plug-in loading method uniformly